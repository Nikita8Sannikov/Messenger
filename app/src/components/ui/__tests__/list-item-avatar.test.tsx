import { render, screen } from "@testing-library/react";

import ListItemAvatarComponent from "@components/ui/list-item-avatar";

describe("ListItemAvatarComponent", () => {
	const mockProps = {
		children: <div>Test Avatar</div>,
		className: "custom-class",
	};

	it("should render with children", () => {
		render(<ListItemAvatarComponent {...mockProps} />);
		const avatar = screen.getByText("Test Avatar");
		expect(avatar).toBeInTheDocument();
	});

	it("should apply custom className", () => {
		render(<ListItemAvatarComponent {...mockProps} />);
		const avatar = screen.getByText("Test Avatar").parentElement;
		expect(avatar).toHaveClass("custom-class");
	});

	it("should render without custom className", () => {
		render(<ListItemAvatarComponent children={mockProps.children} />);
		const avatar = screen.getByText("Test Avatar").parentElement;
		expect(avatar).not.toHaveClass("custom-class");
	});

	it("should render with empty className", () => {
		render(<ListItemAvatarComponent {...mockProps} className="" />);
		const avatar = screen.getByText("Test Avatar").parentElement;
		expect(avatar).not.toHaveClass("custom-class");
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<ListItemAvatarComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
