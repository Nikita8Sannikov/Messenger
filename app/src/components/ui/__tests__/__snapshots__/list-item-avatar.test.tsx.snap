// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemAvatarComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiListItemAvatar-root custom-class css-1ulznza-MuiListItemAvatar-root"
  >
    <div>
      Test Avatar
    </div>
  </div>
</DocumentFragment>
`;
